### PR TITLE
Model performance

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: controller/persistence/base_store.go [hunk 1]
```diff
@@ -165,39 +160,136 @@ func validateEntityIds(tx *bbolt.Tx, store NameIndexedStore, field string, ids [
...
-				semantic = string(semanticValue)
-			}
+		if _, semanticValue := semanticSymbol.Eval(ctx.tx, policyId); semanticValue != nil {
+			semantic = string(semanticValue)
+		}
+		evaluatePolicyAgainstEntity(ctx, semantic, entityId, policyId, ids, roles, entityRoles)
+	}
+}
+
+func (store *baseStore) updateServicePolicyRelatedRoles(ctx *roleAttributeChangeContext, entityId []byte, newRoleAttributes []boltz.FieldTypeAndValue) {
```
Comment (andrewpmartinez) [inline controller/persistence/base_store.go:217]: Seems weird for this to be in a file called `base_store.go`

File: controller/persistence/service_edge_router_policy_store.go [hunk 2]
```diff
@@ -154,16 +160,55 @@ Optimizations
...
 3. Related entity deletes should be handled automatically by FK Indexes on those entities (need to verify the reverse as well/deleting policy)
 */
-func (store *serviceEdgeRouterPolicyStoreImpl) edgeRouterRolesUpdated(ctx *boltz.PersistContext, policy *ServiceEdgeRouterPolicy) {
-	roleIds, err := store.getEntityIdsForRoleSet(ctx.Bucket.Tx(), FieldEdgeRouterRoles, policy.EdgeRouterRoles, policy.Semantic, store.stores.edgeRouter.indexRoleAttributes, store.stores.edgeRouter)
-	if !ctx.Bucket.SetError(err) {
-		ctx.Bucket.SetError(store.edgeRouterCollection.SetLinks(ctx.Bucket.Tx(), policy.Id, roleIds))
+func (store *serviceEdgeRouterPolicyStoreImpl) edgeRouterRolesUpdated(persistCtx *boltz.PersistContext, policy *ServiceEdgeRouterPolicy) {
+	ctx := &roleAttributeChangeContext{
+		tx:                    persistCtx.Bucket.Tx(),
+		roleSymbol:            store.symbolEdgeRouterRoles,
```
Comment (andrewpmartinez) [inline controller/persistence/service_edge_router_policy_store.go:166]: Is it `roleSymbol` or `roleSymbols`?

File: tests/ats-ctrl.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ v: 3
 #  memory:
 #    path: ctrl.memprof
 
-db:                     testdata/ctrl.db
+db:                     testdata/${ZITI_TEST_DB}.db
```
Comment (andrewpmartinez) [inline tests/ats-ctrl.yml:10]: Does this default or is this now a new required env variable to run the ATs?

Comment (andrewpmartinez) [reply tests/ats-ctrl.yml:10]: Mmm I see. `StartServer()` sets it.

File: controller/persistence/base_store.go (hunk 1 above)
Comment (plorenz) [reply controller/persistence/base_store.go]: Moved the funcs that used by multiple policy related stores to policy_common.go

File: controller/persistence/service_edge_router_policy_store.go (hunk 2 above)
Comment (plorenz) [reply controller/persistence/service_edge_router_policy_store.go]: Should actually be rolesSymbol. Fixed :)

**Pull request merged**